### DISCUSSION TITLE
VR Operator pose always empty

### DISCUSSION THREAD

**Discussion by risk-v** [cables-gl/cables_docs#414] [! Welcome - readme first!]
Hi everyone,

I am trying to access the Viewer Pose Object in the VR Operator but it's always empty --> { }
I tested this on the Cables.GL editor with the WebXR Chrome extension and in the Oculus Quest 2. Are you able to access it? Thanks in advance.

Regards,
Esteban

**Comment by pandrr** [comment]
how do you access it ?

**Comment by pandrr** [comment]
javascripts JSON.stringify can not convert this object to a string, not sure why.
so its shown as {}

on dev.cables.gl it now shows at least the class name as an indicator that its not an empty object.

investigating further...

best way to debug is to use the browser debugger and/or console.log and find your properties

**Reply by risk-v** [reply]
Hi @pandrr,

Thanks for looking into this. Actually, I managed to get the translation and rotation values for the three-axis using the transformation matrix. I added new Value Ports to the VR operator and now I can access them easily.

**Comment by pandrr** [comment]
you can also access that and other vr objects in ops after the vr op like this:

```
    cgl.frameStore.xrSession
    cgl.frameStore.xrFrame
    cgl.frameStore.xrViewerPose
    cgl.frameStore.xrReferenceSpace
```

**Reply by risk-v** [reply]
Thanks again @pandrr. That's another feature I love about Cables, the possibility of accessing the operators' code it's amazing! :)
